models/_sector/dex/bot_trades/solana/platforms/consortium_key_solana_bot_trades.sql

```diff
@@ -27,8 +27,14 @@ with
         from {{ source("solana", "account_activity") }} as activity
         join {{ source("solana", "transactions") }} as transactions on tx_id = id
         where
-            activity.block_time >= timestamp '{{project_start_date}}'  -- FirstTradeTimestamp
-            and transactions.block_time >= timestamp '{{project_start_date}}'  -- FirstTradeTimestamp
+            {% if is_incremental() %}
+                {{ incremental_predicate("activity.block_time") }}
+                and {{ incremental_predicate("transactions.block_time") }}
+            {% else %}
+                activity.block_time >= timestamp '{{project_start_date}}'  -- FirstTradeTimestamp
+                and transactions.block_time >= timestamp '{{project_start_date}}'  -- FirstTradeTimestamp
+            {% endif %}
+
```

Comment:
this is one that i tried to add in PR phase, but the use case needs to see full history. it is slow, but we couldn't find a better way around it. ultimately the output is a distinct list of users, but filtering outputs missing data.

i wonder if another option is to query the spell itself with `{{ this }}` and union in the source data with the incremental flag, so that at least getting history of users is faster on the spell

Reply:
here is thread with thoughts:
https://github.com/duneanalytics/spellbook/pull/5880#discussion_r1610273106

Reply:
Good catch, I think I'll just remove it for now but we need some sort of filter here otherwise this will stop working at some point. We can come back to this when we look closer at solana models.